internal/graph/check_test.go

```diff
@@ -50,29 +51,125 @@ type document
 	define viewer: [group#member] or editor
 	define editor: [group#member] and allowed`).TypeDefinitions
 
-	ctx := typesystem.ContextWithTypesystem(context.Background(), typesystem.New(
-		&openfgav1.AuthorizationModel{
-			Id:              ulid.Make().String(),
-			TypeDefinitions: typedefs,
-			SchemaVersion:   typesystem.SchemaVersion1_1,
-		},
-	))
+		ctx := typesystem.ContextWithTypesystem(context.Background(), typesystem.New(
+			&openfgav1.AuthorizationModel{
+				Id:              ulid.Make().String(),
+				TypeDefinitions: typedefs,
+				SchemaVersion:   typesystem.SchemaVersion1_1,
+			},
+		))
+
+		resp, err := checker.ResolveCheck(ctx, &ResolveCheckRequest{
+			StoreID:            storeID,
+			TupleKey:           tuple.NewTupleKey("document:1", "viewer", "user:jon"),
+			ResolutionMetadata: &ResolutionMetadata{Depth: 2},
+		})
+		require.NoError(t, err)
+		require.True(t, resp.Allowed)
 
-	resp, err := checker.ResolveCheck(ctx, &ResolveCheckRequest{
-		StoreID:            storeID,
-		TupleKey:           tuple.NewTupleKey("document:1", "viewer", "user:jon"),
-		ResolutionMetadata: &ResolutionMetadata{Depth: 2},
+		resp, err = checker.ResolveCheck(ctx, &ResolveCheckRequest{
+			StoreID:            storeID,
+			TupleKey:           tuple.NewTupleKey("document:2", "editor", "user:x"),
+			ResolutionMetadata: &ResolutionMetadata{Depth: 2},
+		})
+		require.ErrorIs(t, err, ErrResolutionDepthExceeded)
+		require.Nil(t, resp)
 	})
-	require.NoError(t, err)
-	require.True(t, resp.Allowed)
 
-	resp, err = checker.ResolveCheck(ctx, &ResolveCheckRequest{
-		StoreID:            storeID,
-		TupleKey:           tuple.NewTupleKey("document:2", "editor", "user:x"),
-		ResolutionMetadata: &ResolutionMetadata{Depth: 2},
+	t.Run("exclusion_resolves_deterministically_1", func(t *testing.T) {
+		ds := memory.New()
+
+		storeID := ulid.Make().String()
+
+		err := ds.Write(context.Background(), storeID, nil, []*openfgav1.TupleKey{
+			tuple.NewTupleKey("document:budget", "admin", "user:*"),
+			tuple.NewTupleKeyWithCondition("document:budget", "viewer", "user:maria", "condX", nil),
+		})
+		require.NoError(t, err)
+
+		typedefs := parser.MustTransformDSLToProto(`model
+	schema 1.1
+type user
+
+type document
+  relations
+	define admin: [user:*]
+	define viewer: [user with condX] but not admin
+
+condition condX(x: int) {
+	x < 100
+}
+`).TypeDefinitions
+
+		checker := NewLocalChecker(ds)
+
+		ctx := typesystem.ContextWithTypesystem(context.Background(), typesystem.New(
+			&openfgav1.AuthorizationModel{
+				Id:              ulid.Make().String(),
+				TypeDefinitions: typedefs,
```

Comment:
Here and in the other test you forgot to add `Conditions:      model.Conditions,`

We have this problem elsewhere in our code, too 😓 

https://github.com/openfga/openfga/blob/cfa2b13611f7509d2d58f51bdff3382aee991585/pkg/server/test/list_objects.go#L578-L582

We should find a way of making this kind of sneaky bug not possible... not sure how, though.

Reply:
I suppose the underlying problem is that in our tests we want to pass in an AuthorizationModel that includes a test generated `Id`, but `parser.MustTransformDSLToProto` doesn't generate an `Id` for us. 

It may be a nice helper to have a test utils package in the parser that produces ☝️ . cc'ing @rhamzeh to get his opinion on that idea.

Reply:
Added a helper func and refactored to use it.